### PR TITLE
Ediit READMEs based on Simon's feedback; sync dependencies in conda script

### PR DESCRIPTION
**Pull request by anargyri**
### Description

- Made edits according to [Simon's suggestions](https://github.com/microsoft/simonhzh/blob/main/recommenders-github-repo/misc/reco-utils-setup-confusion.md)
- One of the suggestions was to sync dependencies in generate_conda_file.py with setup.py 
- Part of the edits was instructions for conda and Python 3.7.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Comment by laserprec** [comment]
image
I think the code coverage decrease is not due to any changes in this PR. **Please disregard** if anyone is concerned about this. Please see below you want to know the details 😄:

Currently we have two pipelines running two slightly different tests: the `pr-gate` that runs only the unit tests and the code coverage is around [62%](https://codecov.io/github/microsoft/recommenders/commit/65cd3001f3db9f2738740271da0deb8223676c50) and the `nightly` build that runs the full tests suites: unit, smoke and integration tests. The coverage is around [74%](https://codecov.io/github/microsoft/recommenders/commit/c5029afad5a749b7c0df7b59f7c6fa0900637372). My [last PR](https://github.com/microsoft/recommenders/pull/1503) involved adding the `nightly` build so the PR runs the full nightly build and remembers the higher coverage number instead.

File: docs/requirements.txt [hunk 1]
```diff
@@ -32,3 +32,8 @@ nvidia-ml-py3>=7.352.0
 tensorflow-gpu>=1.15.0,<2
 torch==1.2.0
 fastai>=1.0.46,<2
+databricks_cli>=0.8.6,<1
+pyarrow>=0.8.0,<1.0.0
+pyspark>=2.4.5,<3.0.0
+cmake>=3.18.4.post1
+xlearn==0.40a1
```
Comment (miguelgfierro) [inline docs/requirements.txt:39]: after this is merged to staging a new doc is going to be generated. I'm not sure if pyspark dep is going to break the docs, let's see how it goes

**Review by miguelgfierro — APPROVED** [review]
LGTM

File: docs/requirements.txt (hunk 1 above)
Comment (anargyri) [reply docs/requirements.txt:39]: I tried the compilation of docs and it seems to work.

**Pull request merged**